x-pack/plugins/observability_solution/observability_ai_assistant_app/public/hooks/use_local_storage.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect, useMemo } from 'react';
+
+export function useLocalStorage<T>(key: string, defaultValue: T) {
+  // This is necessary to fix a race condition issue.
+  // It guarantees that the latest value will be always returned after the value is updated
+  const [storageUpdate, setStorageUpdate] = useState(0);
```

Comment:
it'd be great if we have a test or comment for this race condition. I can't remember what it was about to be honest.

One other thing I think we should keep in mind here is that if two components use the same hook with the same key, they won't get updates from each other due to the fact that the storage event is only dispatched in other browsers. That's not a problem here I think, but maybe good to leave a comment at the top of the file explaining this.

Reply:
I've added tests for the hook which include a test for updating the value.